eskipfile/remote_test.go

```diff
@@ -69,107 +67,100 @@ func TestLoadAll(t *testing.T) {
-			Shunt:       false,
-			Backend:     "https://example.com/",
-		}},
+		expected:        eskip.MustParse(fmt.Sprintf("VALID: %v;", routeBody)),
 	},
 	} {
-		s := createTestServer(test.routeContent, test.routeStatusCode)
-		defer s.Close()
+		ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(test.routeStatusCode)
+			io.WriteString(w, test.routeContent)
+		}))
+		defer ts.Close()
 
 		t.Run(test.title, func(t *testing.T) {
-			options := &RemoteWatchOptions{RemoteFile: s.URL, Threshold: 10, Verbose: true, FailOnStartup: true}
+			options := &RemoteWatchOptions{RemoteFile: ts.URL, Threshold: 10, Verbose: true, FailOnStartup: true}
 			client, err := RemoteWatch(options)
-			if err == nil {
-				defer client.(*remoteEskipFile).Close()
-			}
 
-			if err == nil && test.fail {
-				t.Error("failed to fail")
-				return
-			} else if err != nil && !test.fail {
-				t.Error(err)
-				return
-			} else if test.fail {
+			if test.fail {
+				assert.Condition(t, func() bool { return err != nil }, "expected error")
 				return
 			}
 
-			r, err := client.LoadAll()
-			if err != nil {
-				t.Error(err)
-				return
-			}
+			require.NoError(t, err)
 
-			if len(r) == 0 {
-				r = nil
-			}
+			defer client.(*remoteEskipFile).Close()
 
-			if !cmp.Equal(r, test.expected) {
-				t.Errorf("invalid routes received\n%s", cmp.Diff(r, test.expected))
-			}
+			r, err := client.LoadAll()
+			require.NoError(t, err)
+
+			assert.Equal(t, r, test.expected)
 		})
 	}
 }
 
 func TestLoadAllAndUpdate(t *testing.T) {
 	for _, test := range []struct {
-		title               string
-		validRouteContent   string
-		invalidRouteContent string
-		expectedToFail      bool
-		fail                bool
+		title          string
+		content        string
+		contentUpdated string
+		expectedToFail bool
+		fail           bool
 	}{{
-		title:               "Download invalid update and all routes returns routes nil",
-		validRouteContent:   fmt.Sprintf("VALID: %v;", routeBody),
-		invalidRouteContent: fmt.Sprintf("MISSING_SEMICOLON: %v\nVALID: %v;", routeBody, routeBody),
-		expectedToFail:      true,
+		title:          "Download invalid update and all routes returns routes nil",
+		content:        fmt.Sprintf("VALID: %v;", routeBody),
+		contentUpdated: fmt.Sprintf("MISSING_SEMICOLON: %v\nVALID: %v;", routeBody, routeBody),
+		expectedToFail: true,
+	}, {
+		title:          "Download valid update and all routes returns routes",
+		content:        fmt.Sprintf("VALID: %v;", routeBody),
+		contentUpdated: fmt.Sprintf("DIFFERENT_ID: %v;\nVALID: %v;", routeBody, routeBody),
+		expectedToFail: false,
 	},
 	} {
 		t.Run(test.title, func(t *testing.T) {
-			testValidServer := createTestServer(test.validRouteContent, 200)
-			defer testValidServer.Close()
+			ch := make(chan string, 1)
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				routeString := <-ch
```

Comment:
This will work but you have to send response content before each request.
Alternatively you can use e.g. https://pkg.go.dev/sync/atomic#Value and update it when needed

Reply:
I prefer to go with the channels for now to force setting the body.

Reply:
Since we do not have goroutines, we could also just change the content itself, without an issue.
Do I miss anything?

Reply:
test server serves in a separate goroutine